utils/url.tsx

```diff
@@ -122,10 +122,10 @@ export function validateChannelUrl(url: string): React.ReactElement[] {
     const isDirectMessageFormat = directMessageRegex.test(url);
 
     const cleanedURL = cleanUpUrlable(url);
-    const urlMatched = url.match(/[a-z0-9]([-_\w]*)[a-z0-9]/);
+    const urlMatched = url.match(/[a-z0-9]([-_]*[a-z0-9]+)*$/);
```

Comment:
@koox00 Does regex looks good to you.

Reply:
Instead of the repeating group there, you could make the second group optional. I think  `/^[a-z0-9]([a-z0-9\-\_]*[a-z0-9])?$/` would work for this. That also gets rid of the nested `*` groups which I think is what's being caught by the code scanner.

That also matches [the one used by the server](https://github.com/mattermost/mattermost-server/blob/dbcb5f2ffb718c0a252beceef85343e56580f3e0/model/utils.go#L519) more closely, minus its special case for DMs having a double underscore in the middle of it (which is actually redundant, but that's besides the point).

Reply:
@hmhealey Thanks for pointing this out. Will fix in my next commit.